assets/app/view/game/part/city_slot.rb

```diff
@@ -82,7 +82,13 @@ def on_click(event)
           entity = @selected_company || step.current_entity
           actions = step.actions(entity)
           return if (%w[remove_token place_token] & actions).empty?
-          return if @token && !step.can_replace_token?(entity, @token)
+
+          cheaters = []
+          entity.abilities(:token) do |ability|
```

Comment:
ok, i change my mind since you require the first ability, switch it back sorry

Reply:
np, it's not that I need the first so much as I don't know how we'd deal with the situation where one entity has multiples of this ability type; with or without the block syntax this code would need changes once we have a game with that situation anyway